### PR TITLE
Support for pbl scheme

### PR DESCRIPTION
**Pull request by oelbert**
**Description**
This PR adds features needed for the ported PBL scheme. Mostly this means adding relevant namelist switches, but also includes a few useful physics constants and some 4d-field infrastructure

**How Has This Been Tested?**
The new namelist has been tested with the new PBLConfig in my PySHiELD branch, the 4d field allocator has also been used in the PBL scheme testing.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] New check tests, if applicable, are included

### CODE REVIEW

File: ndsl/dsl/typing.py [hunk 1]
```diff
@@ -63,10 +63,19 @@ def global_set_floating_point_precision():
 IntFieldIJ = Field[gtscript.IJ, Int]
 IntFieldK = Field[gtscript.K, Int]
 BoolField = Field[gtscript.IJK, Bool]
+BoolFieldIJ = Field[gtscript.IJ, Bool]
 
 Index3D = Tuple[int, int, int]
 
 
+def set_4d_field_size(n, dtype):
```
Comment (FlorianDeconinck) [inline ndsl/dsl/typing.py:71]: Ping @CharlesKrop, we have a helper function in NDSL to get 4D field types after this PR is merged

**Review by FlorianDeconinck — APPROVED** [review]
LGTM

**Review by fmalatino — APPROVED** [review]
lgtm

File: ndsl/dsl/typing.py (hunk 1 above)
Comment (xyuan) [reply ndsl/dsl/typing.py:71]: for 4D array, do we still keep k as vertical direction? so the verticalLoop node is the same as before, and horizontal execution nodes contain 3d loops?

**Comment by oelbert** [comment]
@bensonr the PhysicsConfig dataclass should contain all of the parameters for the physics routines, but each scheme should have a child class that contains the parameters it needs, so there will be a PBLConfig, a MicrophysicsConfig, etc. Right now NDSL still has the infrastructure for parsing the namelists into Pace/PyFV3/PySHiELD, but I'd like to get some of these pjysics schemes in before tackling https://github.com/NOAA-GFDL/NDSL/issues/64

**Review by xyuan — APPROVED** [review]

**Comment by FlorianDeconinck** [comment]
@xyuan Data dimensions (anything after I, J, K) are executed serially.

**Review by bensonr — APPROVED** [review]

**Pull request merged**